Bottom: 90d1bcde626d7f02a3c611e9797aa14dd02d4ffd
Top:    3c4ae25a2ff3dd2a6499e237020cdc303b201b69
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-01-07 02:19:11 +0000

Fix to ParseGroupString()


---

diff --git a/src/misc.c b/src/misc.c
index 040124c..545aaac 100644
--- a/src/misc.c
+++ b/src/misc.c
@@ -1136,7 +1136,14 @@ ParseGroupString (char *group_string, LayerGroupType *LayerGroup, int *LayerN)
           /* ignore white spaces and check for separator */
           while (*s && isspace ((int) *s))
             s++;
-          if (!*s || *s == ':')
+
+          /* Backup so the outer loop terminates on the NULL termination */
+          if (*s == '\0') {
+            s--;
+            break;
+          }
+
+          if (*s == ':')
             break;
         }
       LayerGroup->Number[group] = member;
